open-api/rest-catalog-open-api.yaml

```diff
@@ -3324,6 +3348,184 @@ components:
           type: integer
           format: int64
 
+    BooleanTypeValue:
+      type: boolean
+
+    IntegerTypeValue:
+      type: integer
+
+    LongTypeValue:
+      type: integer
+      format: int64
+
+    FloatTypeValue:
+      type: number
+      format: float
+
+    DoubleTypeValue:
+      type: number
+      format: double
+
+    DecimalTypeValue:
+      type: string
+      description:
+        "Decimal types are serialized as a string to preserve exact numeric precision"
```

Comment:
I think we need to be more specific for how the string is serialized. For example for decimal, we should describe this logic:

```
BigDecimal decimalValue = (BigDecimal) defaultValue;
        if (decimalValue.scale() >= 0) {
          generator.writeString(decimalValue.toPlainString());
        } else {
          generator.writeString(decimalValue.toString());
        }
```

Reply:
Maybe also add examples for each type value